File: online-judge-solutions/Codeforces/1167A.c
//https://codeforces.com/problemset/problem/1167/A
//Lang		:	GNU C11
//Time		:	15 ms
//Memory	:	0 KB
#include<stdio.h>
#include<string.h>
int main(){
	int t,n;
	scanf("%d",&t);
	while(t--){
		char s[100];
		scanf("%d%s",&n,s);
		int i=-1;
		for(int j=0;j<=strlen(s) && i<0;j++){
			if(s[j]=='8'){i=j;}
		}
		(i>=0 && n-i>=11) ? printf("YES\n"):printf("NO\n");
	}
}

File: online-judge-solutions/Codeforces/1167A.py
#https://codeforces.com/problemset/problem/1167/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
t=int(input())
for x in range(0,t):
    n,s=int(input()),input()
    i=s.find('8')
    print("YES") if (i>=0 and n-i>=11) else print("NO")
